### PR TITLE
Update git_pillar docs, add git.list_worktrees function

### PR DESCRIPTION
**Pull request by terminalmage**

### CODE REVIEW

**Comment by jfindlay** [comment]
@terminalmage, there is a unit test that needs to be updated as well.

**Comment by terminalmage** [comment]
@jfindlay Fixed, thanks. I thought I looked at that one already, but I must have been confusing it with something else. Thanks for the catch.

File: salt/modules/git.py [hunk 1]
```diff
@@ -1548,6 +1546,103 @@ def list_tags(cwd, user=None, ignore_retcode=False):
...
+    if worktree_root.startswith('.git'):
+        worktree_root = os.path.join(cwd, worktree_root)
+    if not os.path.isdir(worktree_root):
+        return {}
+
+    worktree_info = {}
+    for worktree_name in os.listdir(worktree_root):
+        gitdir_file = os.path.join(worktree_root, worktree_name, 'gitdir')
+        try:
+            with open(gitdir_file, 'r') as fp_:
```
Comment (jfindlay) [inline salt/modules/git.py]: This should be a `salt.utils.fopen`.  Everything looks great otherwise.

**Comment by terminalmage** [comment]
@jfindlay fixed that call to ``open()``, thanks.

**Pull request merged**